vm/src/obj/objset.rs

```diff
@@ -39,12 +64,40 @@ fn set_add(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
     );
     let mut mut_obj = s.borrow_mut();
 
-    if let PyObjectPayload::Set { ref mut elements } = mut_obj.payload {
-        let key = item.get_id();
-        elements.insert(key, item.clone());
-        Ok(vm.get_none())
-    } else {
-        Err(vm.new_type_error("set.add is called with no list".to_string()))
+    match mut_obj.payload {
```

Comment:
The `if let` syntax is better in this case of a match with a single arm and a catch all. At least, this is what clippy suggests.